nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/DatasetType.java

```diff
@@ -0,0 +1,78 @@
+package org.nmrfx.processor.datasets;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public enum DatasetType {
+    NMRFX(".nv"),
+    UCSF(".ucsf"),
+    SPINit("data.dat");
+
+    String extension;
+    DatasetType(String extension) {
+        this.extension = extension;
+    }
+
+    /**
+     * Determine the type of the file from the fileName.
+     * @param file the file to check
+     * @return an Optional containing DatasetType if fileName is valid file or empty if not.
+     */
+    public static Optional<DatasetType> typeFromName(File file) {
+        String name = file.getName();
+        for (var ext:values()) {
+            if (name.endsWith(ext.extension)) {
+                return Optional.of(ext);
+            }
+        }
+        return Optional.empty();
```

Comment:
Duplication of typeFromName(String ..), and more importantly, doesn't detect directories containing header+data as RS2D files, but that's what's passed to open().

Reply:
Fixed.